src/Common/tests/TestUtilities/System/PlatformDetection.Windows.cs

```diff
@@ -1,294 +1,38 @@
-        public static bool IsCentos6 => false;
-        public static bool IsOpenSUSE => false;
-        public static bool IsUbuntu => false;
-        public static bool IsDebian => false;
-        public static bool IsAlpine => false;
-        public static bool IsDebian8 => false;
-        public static bool IsUbuntu1404 => false;
-        public static bool IsUbuntu1604 => false;
-        public static bool IsUbuntu1704 => false;
-        public static bool IsUbuntu1710 => false;
-        public static bool IsUbuntu1710OrHigher => false;
-        public static bool IsUbuntu1804 => false;
-        public static bool IsUbuntu1810OrHigher => false;
-        public static bool IsTizen => false;
-        public static bool IsNotFedoraOrRedHatFamily => true;
-        public static bool IsFedora => false;
-        public static bool IsWindowsNanoServer => (IsNotWindowsIoTCore && GetInstallationType().Equals("Nano Server", StringComparison.OrdinalIgnoreCase));
-        public static bool IsWindowsServerCore => GetInstallationType().Equals("Server Core", StringComparison.OrdinalIgnoreCase);
-        public static int WindowsVersion => GetWindowsVersion();
-        public static bool IsMacOsHighSierraOrHigher { get; } = false;
-        public static Version ICUVersion => new Version(0, 0, 0, 0);
-        public static bool IsRedHatFamily => false;
-        public static bool IsNotRedHatFamily => true;
-        public static bool IsRedHatFamily6 => false;
-        public static bool IsRedHatFamily7 => false;
-        public static bool IsNotRedHatFamily6 => true;
-        public static bool SupportsSsl3 => GetSsl3Support();
-
-        public static bool IsWindows10Version1607OrGreater => 
-            GetWindowsVersion() == 10 && GetWindowsMinorVersion() == 0 && GetWindowsBuildNumber() >= 14393;
-        public static bool IsWindows10Version1703OrGreater => 
-            GetWindowsVersion() == 10 && GetWindowsMinorVersion() == 0 && GetWindowsBuildNumber() >= 15063;
-        public static bool IsWindows10Version1709OrGreater => 
-            GetWindowsVersion() == 10 && GetWindowsMinorVersion() == 0 && GetWindowsBuildNumber() >= 16299;
-
-        // Windows OneCoreUAP SKU doesn't have httpapi.dll
-        public static bool IsNotOneCoreUAP =>  
-            File.Exists(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "System32", "httpapi.dll"));
-
-        public static bool IsWindowsIoTCore
-        {
-            get
+        //
+        // Do not use the " { get; } = <expression> " pattern here. Having all the initialization happen in the type initializer
+        // means that one exception anywhere means all tests using PlatformDetection fail. If you feel a value is worth latching,
+        // do it in a way that failures don't cascade.
+        //
+
+        public static bool IsWindows => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```

Comment:
It's good that use runtime detection here. I think we should collapse the three different PlatformDetection files into a single one and always use runtime detection. Runtime detection is also what we do in dotnet/runtime: https://github.com/dotnet/runtime/blob/99cfd79e7c8e8d42bc2c55f6662d64c74cbe8428/src/libraries/Common/tests/TestUtilities/TestUtilities.csproj#L33-L36